### PR TITLE
Fixes #3507 - Allow Pulp streamer to log to console

### PR DESCRIPTION
**Pull request by ehelms**
Switches Pulp streamer to use the base Pulp server logging configuration
methods.

### CODE REVIEW

**Comment by dralley** [comment]
The test that failed looks unrelated to this PR

```
======================================================================
ERROR: test_get_certs_from_string_empty (test_repo_cert_utils.TestCertVerify)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins/workspace/unittest-pulp-pr/node-type/f27-np/pulp/repoauth/test/test_repo_cert_utils.py", line 392, in test_get_certs_from_string_empty
    certs = self.utils.get_certs_from_string("")
  File "/home/jenkins/workspace/unittest-pulp-pr/node-type/f27-np/pulp/repoauth/pulp/repoauth/repo_cert_utils.py", line 358, in get_certs_from_string
    bio = BIO.MemoryBuffer(data)
  File "/usr/lib64/python2.7/site-packages/M2Crypto/BIO.py", line 191, in __init__
    m2.bio_write(self.bio, data)
BIOError: padding check failed
```

File: streamer/usr/share/pulp/wsgi/streamer.tac [hunk 1]
```diff
@@ -16,40 +16,10 @@ from pulp.plugins.loader import api as plugin_api
...
-    formatter = logging.Formatter(LOG_FORMAT_STRING)
-    handler.setFormatter(formatter)
-    root_logger.handlers = []
-    root_logger.addHandler(handler)
-
 
 # Initialization of application dependencies.
 streamer_config = load_configuration(DEFAULT_CONFIG_FILES)
-start_logging(streamer_config)
+start_logging()
```
Comment (dparalen) [inline streamer/usr/share/pulp/wsgi/streamer.tac]: `start_logging` doesn't use this modules `LOG_FORMAT_STRING` which makes the logs of `streamer` indistinguishable by prefix from `server` logs.

**Comment by ehelms** [comment]
Given the LOG_FORMAT_STRING is defined after pulp.server.logs the constant in streamer.tac takes precedence as I understand python thereby setting the logging format with no regression. Testing this locally, I still see the following:

```
Mar 23 00:25:26 localhost pulp_streamer: [-] Log opened.
Mar 23 00:25:26 localhost pulp_streamer: [-] twistd 12.2.0 (/usr/bin/python 2.7.5) starting up.
Mar 23 00:25:26 localhost pulp_streamer: [-] reactor class: twisted.internet.epollreactor.EPollReactor.
Mar 23 00:25:26 localhost pulp_streamer: [-] Site starting on 8751
Mar 23 00:25:26 localhost pulp_streamer: [-] Starting factory <twisted.web.server.Site instance at 0x4b68050>
```

File: streamer/usr/share/pulp/wsgi/streamer.tac [hunk 2]
```diff
@@ -16,40 +16,10 @@ from pulp.plugins.loader import api as plugin_api
...
-LOG_PATH = os.path.join('/', 'dev', 'log')
-
-
-def start_logging(config):
-    """
-    Configure the Pulp streamer syslog handler for the configured log level.
-    """
-    # Get and set up the root logger with our configured log level
-    try:
-        log_level = config.get('streamer', 'log_level')
```
Comment (dparalen) [inline streamer/usr/share/pulp/wsgi/streamer.tac]: Oops, `streamer` `log_level` gone

File: streamer/usr/share/pulp/wsgi/streamer.tac [hunk 3]
```diff
@@ -16,40 +16,10 @@ from pulp.plugins.loader import api as plugin_api
 
 DEFAULT_LOG_LEVEL = logging.INFO
 LOG_FORMAT_STRING = 'pulp_streamer: %(name)s:%(levelname)s: %(message)s'
-LOG_PATH = os.path.join('/', 'dev', 'log')
```
Comment (dparalen) [inline streamer/usr/share/pulp/wsgi/streamer.tac]: this is odd; evaluates to `dev/log` ??

**Comment by dparalen** [comment]
@ehelms I didn't run the code, obviously O:-) But I'd say relying on constants being pulled in by the interpreter in particular order might not be wise. Another thing is the `loglevel` setting is removed now too. I'd still recommend adding the logging stream configuration option over removing the whole logging initialization or at least refactoring the server logging initialization to accept some explicit values.

**Comment by ehelms** [comment]
@dparalen I am still not sure I follow where the regression is introduced given this largely just moves initialization and some configuration to the base Pulp application. I still feel like a user can set what's needed through that for the entire application. Otherwise I have to duplicate a chunk of code.

**Comment by dralley** [comment]
@bmbouter @ehelms ping, ^^

**Comment by ipanova** [comment]
closing this PR since the bug is closed as wont-fix https://pulp.plan.io/issues/3507

**Pull request closed without merging**